src/Features/Core/Portable/Completion/Providers/AbstractInternalsVisibleToCompletionProvider.cs

```diff
@@ -0,0 +1,218 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Completion.Providers
+{
+    internal abstract class AbstractInternalsVisibleToCompletionProvider : CommonCompletionProvider
+    {
+        private const string ProjectGuidKey = nameof(ProjectGuidKey);
+
+        protected abstract IImmutableList<SyntaxNode> GetAssemblyScopedAttributeSyntaxNodesOfDocument(SyntaxNode documentRoot);
+        protected abstract SyntaxNode GetConstructorArgumentOfInternalsVisibleToAttribute(SyntaxNode internalsVisibleToAttribute);
+
+        internal override bool IsInsertionTrigger(SourceText text, int insertedCharacterPosition, OptionSet options)
+        {
+            var ch = text[insertedCharacterPosition];
+            return ch == '\"';
```

Comment:
I would trigger if the user types doublequote or if they are typing letters after the double quote. You're already checking the first thing. For the other, we have a method `CommonCompletionUtilities.IsStartingNewWord`. It takes an `isWordStart` delegate which you could have be `SyntaxFacts.IsIdentifierStartCharacter(c) || c == '"'`. For the `isWordCharacter` delegate, you could pass in the language specific helpers for `IsWordCharacter` in CompletionUtilities.cs/vb. Make sense?

Basically, I want you to trigger for:
[InternalsVisibleTo("$$
and
[InternalsVisible("My$$
where the user starts modifying the contents of the string without typing the opening quote.